ament_tools/build_types/ament_cmake.py

```diff
@@ -99,6 +102,12 @@ def on_build(self, context):
                 extra_cmake_args += ['-DAMENT_CMAKE_SYMLINK_INSTALL=1']
             extra_cmake_args += context.cmake_args
             extra_cmake_args += context.ament_cmake_args
+        if context.ninja_build:
+            extra_cmake_args += ['-G']
+            extra_cmake_args += ['Ninja']
+            extra_cmake_args += ["-DCMAKE_MAKE_PROGRAM=/usr/bin/ninja"]
```

Comment:
Do you have to specify this path? Doesn't the Ninja generator automatically use `ninja` as the make program?

Same below. This should be moved into a function which can be called from both locations.